### PR TITLE
rust analyzer snake_case fix #2882

### LINKED ISSUE #2882: Generated constants should be UPPER_SNAKE_CASE
### Describe the Bug

Generated constants are not upper snake case and this triggers warnings from the `rust-analyzer`.

### Steps to Reproduce

Save this code to a `.rs` file:
```
#[wasm_bindgen]
extern "C" {
    pub type Something;
}
```

Import the required mods and run the `rust-analyzer`.

### Expected Behavior

No warnings should appear.

### Actual Behavior

The following error appears:
```
Diagnostic { code: DiagnosticCode("incorrect-ident-case"), message: "Constant `__wbg_generated_const_Something` should have UPPER_SNAKE_CASE name, e.g. `__WBG_GENERATED_CONST_SOMETHING`", range: 732..747, severity: WeakWarning, unused: false, experimental: false, fixes: Some([Assist { id: AssistId("change_case", QuickFix), label: "Rename to __WBG_GENERATED_CONST_SOMETHING", group: None, target: 732..747, source_change: None }]) }
```
image

### Additional Context

I've tried to disable the warning by using `#[allow(non_snake_case,non_upper_case_globals)]` but it does not do the trick. I've also tried to tweak the `rust-analyzer` to ignore this but I also failed.

### PR DESCRIPTION
**Pull request by eli-front**
Fixes #2882

### CODE REVIEW

**Review by Liamolucko — CHANGES_REQUESTED** [review]

File: crates/backend/src/codegen.rs [hunk 1]
```diff
@@ -1104,7 +1118,7 @@ impl TryToTokens for ast::ImportFunction {
             quote! {
                 #[cfg(all(target_arch = "wasm32", not(target_os = "emscripten")))]
                 #(#attrs)*
-                #[link(wasm_import_module = "__wbindgen_placeholder__")]
+                #[link(wasm_import_module = "__WBINDGEN_PLACEHOLDER__")]
```
Comment (Liamolucko) [inline crates/backend/src/codegen.rs:1121]: This also needs to remain lowercase.

File: crates/backend/src/codegen.rs [hunk 2]
```diff
@@ -1334,9 +1348,15 @@ impl<'a, T: ToTokens> ToTokens for Descriptor<'a, T> {
             return;
         }
 
-        let name = Ident::new(&format!("__wbindgen_describe_{}", ident), ident.span());
+        let name = Ident::new(
+            &format!("__WBINDGEN_DESCRIBE_{}", ident.to_string().to_uppercase()),
+            ident.span(),
+        );
```
Comment (Liamolucko) [inline crates/backend/src/codegen.rs:1354]: The CLI also relies on this being lowercase.

```suggestion
        let name = Ident::new(&format!("__wbindgen_describe_{}", ident), ident.span());
```

File: crates/test/src/rt/browser.rs [hunk 3]
```diff
@@ -17,7 +17,7 @@ pub struct Browser {
 #[wasm_bindgen]
 extern "C" {
     type HTMLDocument;
-    static document: HTMLDocument;
+    static DOCUMENT: HTMLDocument;
```
Comment (Liamolucko) [inline crates/test/src/rt/browser.rs:20]: This still needs to be lowercase on the JS side:

```suggestion
    #[wasm_bindgen(js_name = document)]
    static DOCUMENT: HTMLDocument;
```

**Comment by eli-front** [comment]
Thanks to @Liamolucko We have passed all CI tests!

**Comment by FrankReh** [comment]
Any maintainers can review this?

**Comment by Liamolucko** [comment]
Sorry, I completely let this slip. In the meantime, though, I found out that it's actually possible to not give consts names at all, by naming them `_`. I think that's a better solution, since it also makes it impossible for the consts to cause name collisions, so I opened #3198 which does that instead.

**Comment by eli-front** [comment]
> Sorry, I completely let this slip. In the meantime, though, I found out that it's actually possible to not give consts names at all, by naming them `_`. I think that's a better solution, since it also makes it impossible for the consts to cause name collisions, so I opened #3198 which does that instead.

What was the original value for naming the constants?

**Comment by Liamolucko** [comment]
> What was the original value for naming the constants?

I don't think there was any, I think it was just that nobody knew you didn't have to give them names.

**Pull request closed without merging**